core/Pos/Crypto/AsBinary.hs

```diff
@@ -44,14 +44,14 @@ checkLenImpl action name expectedLen len
 
 #define Ser(B, Bytes, Name) \
   instance (Bi B, Bi (AsBinary B)) => AsBinaryClass B where {\
-    asBinary = AsBinary . checkLen "asBinary" Name Bytes . encode ;\
-    fromBinary = decodeFull . checkLen "fromBinary" Name Bytes . encode }; \
+    asBinary = AsBinary . checkLen "asBinary" Name Bytes . serialize' ;\
+    fromBinary = decodeFull . checkLen "fromBinary" Name Bytes . getAsBinary }; \
```

Comment:
Speaking of `AsBinary`, I think it should be changed too. Previously (at least when we were using `binary`, not sure about `store`) the idea was that `AsBinary` carries serialized into fixed-size `ByteString` value and its `Bi` instance just puts this `ByteString` w/o any overhead (and this overhead shouldn't be needed, because the value is already serialized). However, with CBOR it doesn't work this way, because if we have `ByteString` we can only serialize it with some overhead. AFAIU, it is not possible to put a fixed-size `ByteString` directly.
What I am thinking about is that we can put `Tokens` inside `AsBinary` instead. In this case we should be able to avoid overhead imposed by `AsBinary`.
Does it make sense? I am not sure, maybe I am saying nonsense.